cmd/zed/manage/lakemanage/lakemanage.go

```diff
@@ -0,0 +1,211 @@
+package lakemanage
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/brimdata/zed/api"
+	"github.com/brimdata/zed/api/client"
+	lakeapi "github.com/brimdata/zed/lake/api"
+	"github.com/segmentio/ksuid"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+func Update(ctx context.Context, lk lakeapi.Interface, conf Config, logger *zap.Logger) error {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+	if err := conf.Index.getRules(ctx, lk); err != nil {
+		return err
+	}
+	branches, err := getBranches(ctx, conf, lk, logger)
+	if err != nil {
+		return err
+	}
+	group, ctx := errgroup.WithContext(ctx)
+	for _, branch := range branches {
+		branch := branch
+		group.Go(func() error {
+			for _, task := range branch.tasks {
+				if _, err := task(ctx); err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+	}
+	return group.Wait()
+}
+
+func Monitor(ctx context.Context, conn *client.Connection, conf Config, logger *zap.Logger) error {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+	timer := time.NewTimer(0)
+	<-timer.C
+	defer func() {
+		if !timer.Stop() {
+			<-timer.C
+		}
+	}()
+	for {
+		switch err := runMonitor(ctx, conf, conn, logger); {
+		case errors.Is(err, syscall.ECONNREFUSED):
+			logger.Info("cannot connect to lake, retrying in 5 seconds")
+		case err != nil:
+			return err
+		}
+		timer.Reset(time.Second * 5)
+		select {
+		case <-timer.C:
```

Comment:
I don't think a time.Timer is needed here since efficiency isn't a concern.
```suggestion
		case <-time.After(5 * time.Second):
```

Reply:
Good point